### PR TITLE
Add Algolia

### PR DESCRIPTION
**Pull request by Haroenv**
I was doubting to add `requires-approval` or not, because default free plan is 10k records, 100k ops; default OSS is 50k / 100k and more needs approval but is definitely possible. Docsearch is also always free for OSS

Thank you for taking the time to work on a PR for Awesome-Open-Source-Supporters!

To ensure your PR is dealt with swiftly please check the following:

- [x] Your submissions are formatted according to the guidelines: ``[Company Name](link) `tag` - description``
- [x] Your additions are ordered alphabetically.
- [x] Your additions are commercial software that offer their services for free to Open Source projects.
- [x] Your additions contain any relevant tags: `requires-approval`
- [x] You have searched the repository for any relevant issues or PRs.
- [x] Any category you are creating has the minimum requirement of 2 items.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,7 @@ A curated list of awesome companies that offer their tools and services for free
 
 ## Miscellaneous
 
+- [Algolia](https://www.algolia.com) - Search as a service API
```
Comment (zachflower) [inline README.md]: Thanks for the submission! I had no idea Algolia offered an OSS tier! While they do have a free tier, their Open Source specific one does require approval, so I think a `requires-approval` tag is appropriate here.

**Review by zachflower — CHANGES_REQUESTED** [review]

**Comment by Haroenv** [comment]
updated @zachflower

**Comment by zachflower** [comment]
Looks good, thanks!

**Comment by joshed-io** [comment]
@Haroenv small correction, the [Algolia for Open Source plan](https://www.algolia.com/for-open-source) is 100k records and 200k ops by default.

**Pull request merged**